src/foundation.cs

```diff
@@ -292,51 +300,60 @@ partial interface NSAttributedString : NSCoding, NSMutableCopying, NSSecureCodin
 		// New API in iOS9 with same signature as an older alternative.
 		// We expose only the *new* one for the new platforms as the old
 		// one was moved to `NSDeprecatedKitAdditions (NSAttributedString)`
+		[NoMac]
 		[iOS (9,0)]
 		[Internal]
 		[Export ("initWithURL:options:documentAttributes:error:")]
 		IntPtr InitWithURL (NSUrl url, [NullAllowed] NSDictionary options, out NSDictionary resultDocumentAttributes, ref NSError error);
+
 		// but we still need to allow the API to work before iOS 9.0
 		// and to compleify matters the old one was deprecated in 9.0
+		[NoMac]
```

Comment:
I think this one should take [NoTV][NoWatch] as well?

Reply:
Good catch. I missed it since I tried to fix https://github.com/xamarin/xamarin-macios/issues/14489 twice before giving up.